### PR TITLE
Cover Cart's related classes with unit tests

### PR DESCRIPTION
**Pull request by stanislav-kozorez**
- [x] Unit tests for CartsController
- [x] Unit tests for CartService
- [x] Unit tests for CartManagerV2
- [x] Adjust Asserts and Recieved method

### CODE REVIEW

File: src/Foundation/Commerce/tests/Services/Cart/CartServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+        [Theory]
+        [InlineData(null, null)]
+        [InlineData(null, "1")]
+        [InlineData("1", null)]
+        public void UpdateCartLine_IfParameterIsNull_ShouldThrowArgumentNullException(string productId, string variantId)
+        {
+            // act & assert
+            Assert.Throws<ArgumentNullException>(
+                () => this.cartService.UpdateCartLine(productId, variantId, this.fixture.Create<decimal>()));
```
Comment (evgenii-nikulin) [inline src/Foundation/Commerce/tests/Services/Cart/CartServiceTests.cs]: I think we should not do `act` and `assert` at the same time.
For check exceptions we can use `[ExpectedException(typeof(ArgumentNullException))]` attribute or `Record.Exception` method to specific information from it.

Comment (stanislav-kozorez) [reply src/Foundation/Commerce/tests/Services/Cart/CartServiceTests.cs]: I have doubts that such approach will bring us any benefits

Comment (stanislav-kozorez) [reply src/Foundation/Commerce/tests/Services/Cart/CartServiceTests.cs]: https://xunit.net/docs/comparisons#note1

File: src/Feature/Checkout/tests/Controllers/CartsControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        public void GetCart_ShouldCallExecuteMethod()
+        {
+            // act
+            this.controller.GetCart();
+
+            // assert
+            this.controller.Received(1).Execute(this.cartService.GetCart);
+        }
+
+        // TODO: Tests are valid. Should be uncommented after fixing issue with Quantity validation
```
Comment (larionovadaria) [inline src/Feature/Checkout/tests/Controllers/CartsControllerTests.cs]: Ok, thanks. Please do not forget add this note to Front-end refactoring task about quantity.

Comment (stanislav-kozorez) [reply src/Feature/Checkout/tests/Controllers/CartsControllerTests.cs]: Already did

**Pull request merged**